pkg/elevator/internal/util/util.go

```diff
@@ -76,3 +85,49 @@ func VersionExposeConfigForVersion(
 	}
 	return catalogv1alpha1.VersionExposeConfig{}, false
 }
+
+func FormDefaults(defaultField *runtime.RawExtension) (defaults map[string]interface{}, err error) {
+	if defaultField != nil {
+		defaults = make(map[string]interface{})
+		if err := yaml.Unmarshal(defaultField.Raw, &defaults); err != nil {
+			return nil, fmt.Errorf("defaults yaml marshalling: %w", err)
+		}
+	}
+	return
+}
+
+func BuildProviderObj(tenantObj *unstructured.Unstructured, providerObj *unstructured.Unstructured, scheme *runtime.Scheme, elevateFields []catalogv1alpha1.FieldPath, defaults interface{}) error {
+	providerObj.SetName(tenantObj.GetName())
+	providerObj.SetNamespace(tenantObj.GetNamespace())
+
+	// controller reference without UID is invalid!
+	// it will fail upon creation
+	if tenantObj.GetUID() != "" {
+		err := controllerutil.SetControllerReference(
```

Comment:
I don't think it is wise to set the controller reference within this function as it's shared with the webhook.
From my perspective the controller reference should only be touched by the controller and it should be explicit and not hidden in a utility function.